sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/KalixServer.scala

```diff
@@ -0,0 +1,223 @@
+
+    addIncludeFilter(KalixComponentTypeFilter)
+
+    def findKalixComponents(packageToScan: String): Seq[BeanDefinition] = {
+      this
+        .findCandidateComponents(packageToScan)
+        .asScala
+        .map { bean =>
+          // by default, the provider set them all as singletons,
+          // we need to make them all a prototype
+          bean.setScope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+
+          // making it only wireable by constructor will simplify our lives
+          // we can review it later, if needed
+          bean.asInstanceOf[AbstractBeanDefinition].setAutowireMode(AbstractBeanDefinition.AUTOWIRE_CONSTRUCTOR)
+          bean
+
+        }
+        .toSeq
+    }
+  }
+
+  class ActionCreationContextFactoryBean(loco: ThreadLocal[ActionCreationContext])
+      extends FactoryBean[ActionCreationContext] {
+
+    // ActionCreationContext is a singleton, so strictly speaking this could return 'true'
+    // However, we still need the ThreadLocal hack to let Spring have access to it.
+    // and we don't want to give it direct access to it, because the impl is private (and better keep it so).
+    // because the testkit uses another ActionCreationContext impl. Therefore we want it to be defined at runtime.
+    override def isSingleton: Boolean = false
+
+    override def getObject: ActionCreationContext = loco.get()
+    override def getObjectType: Class[_] = classOf[ActionCreationContext]
+  }
+
+  class ValueEntityContextFactoryBean(loco: ThreadLocal[ValueEntityContext]) extends FactoryBean[ValueEntityContext] {
+    override def isSingleton: Boolean = false // never!!
+    override def getObject: ValueEntityContext = loco.get()
+    override def getObjectType: Class[_] = classOf[ValueEntityContext]
+  }
+
+  class ViewCreationContextFactoryBean(loco: ThreadLocal[ViewCreationContext])
+      extends FactoryBean[ViewCreationContext] {
+    override def isSingleton: Boolean = false // never!!
+    override def getObject: ViewCreationContext = loco.get()
+    override def getObjectType: Class[_] = classOf[ViewCreationContext]
+  }
+}
+
+@Component
+class KalixServer(
+    applicationContext: ApplicationContext,
+    threadLocalActionContext: ThreadLocal[ActionCreationContext],
+    threadLocalValueEntityContext: ThreadLocal[ValueEntityContext],
+    threadLocalViewContext: ThreadLocal[ViewCreationContext]) {
+
+  val logger: Logger = LoggerFactory.getLogger(getClass)
+
+  val kalixBeanFactory = new DefaultListableBeanFactory(applicationContext)
+
+  @PostConstruct
+  def init(): Unit = {
+    logger.info("Starting Kalix Server!")
+
+    // TODO: it should not be allowed to annotate Kalix components with Spring stereotypes
+    //  otherwise it will be possible to inject kalix components everywhere and that won't work as expected
+    // therefore, we should check all beans in ApplicationContext and fail if any of them implements a Kalix component
```

Comment:
I still need to prove that it will be possible to do this kinds of blockade. I hope we can.